crash when importing aws_route_table

Terraform v0.7.0 crashes when i try to import aws_route_table resource:
```terraform import aws_route_table.private <my_route_table_id>```

crash.log:
https://gist.github.com/drscre/1b7994cadb588924ce627b2487e985a9

I run into this exact problem when trying to import a route table that uses s3 endpoints.

Here is what my route table looks like:

<img width="544" alt="screen shot 2016-08-18 at 1 43 34 pm" src="https://cloud.githubusercontent.com/assets/523384/17784390/5e1c4972-654a-11e6-9154-17325646aa53.png">

```
$ terraform import aws_route_table.public rtb-473fb622
aws_route_table.public: Importing from ID "rtb-473fb622"...
Error importing: 1 error(s) occurred:

* import aws_route_table.public (id: rtb-473fb622): unexpected EOF
panic: runtime error: invalid memory address or nil pointer dereference
2016/08/18 13:45:42 [DEBUG] plugin: terraform: [signal 0xb code=0x1 addr=0x0 pc=0x55d526]
2016/08/18 13:45:42 [DEBUG] plugin: terraform:
2016/08/18 13:45:42 [DEBUG] plugin: terraform: goroutine 104 [running]:
2016/08/18 13:45:42 [DEBUG] plugin: terraform: panic(0x2fb1ba0, 0xc8200120e0)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/go/src/runtime/panic.go:481 +0x3e6
2016/08/18 13:45:42 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.routeIDHash(0xc820460d80, 0xc82062eff0, 0x0, 0x0)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_route.go:362 +0xa6
2016/08/18 13:45:42 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsRouteTableImportState(0xc8204600c0, 0x253e380, 0xc820492000, 0x0, 0x0, 0x0, 0x0, 0x0)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/import_aws_route_table.go:48 +0xf57
2016/08/18 13:45:42 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).ImportState(0xc8203f1a40, 0xc820472640, 0xc8205c5de0, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:254 +0x496
2016/08/18 13:45:42 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).ImportState(0xc820140ce0, 0xc820662880, 0xc820662ba0, 0x0, 0x0)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:493 +0x86
2016/08/18 13:45:42 [DEBUG] plugin: terraform: reflect.Value.call(0x2c6c4c0, 0x3650678, 0x13, 0x37c3bf8, 0x4, 0xc8206bded8, 0x3, 0x3, 0x0, 0x0, ...)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:435 +0x120d
2016/08/18 13:45:42 [DEBUG] plugin: terraform: reflect.Value.Call(0x2c6c4c0, 0x3650678, 0x13, 0xc8206bded8, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:303 +0xb1
2016/08/18 13:45:42 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc820360500, 0xc8203604c0, 0xc8203581b8, 0xc820370300, 0xc820362140, 0x2540cc0, 0xc820662880, 0x16, 0x2540d20, 0xc820662ba0, ...)
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:383 +0x1c2
2016/08/18 13:45:42 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/08/18 13:45:42 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:477 +0x49d
2016/08/18 13:45:42 [ERROR] root: eval: *terraform.EvalImportState, err: import aws_route_table.public (id: rtb-473fb622): unexpected EOF
2016/08/18 13:45:42 [ERROR] root: eval: *terraform.EvalSequence, err: import aws_route_table.public (id: rtb-473fb622): unexpected EOF
2016/08/18 13:45:42 [TRACE] [walkImport] Exiting eval tree: aws_route_table.public (import id: rtb-473fb622)
2016/08/18 13:45:42 [DEBUG] vertex provider.aws (close), got dep: aws_route_table.public (import id: rtb-473fb622)
2016/08/18 13:45:42 [DEBUG] plugin: /usr/local/bin/terraform: plugin process exited
2016/08/18 13:45:42 [DEBUG] plugin: waiting for all plugin processes to complete...



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/builtin/providers/aws/import_aws_route_table.go
+++ b/builtin/providers/aws/import_aws_route_table.go
@@ -40,6 +40,12 @@ func resourceAwsRouteTableImportState(
 				continue
 			}
 
+			if route.DestinationPrefixListId != nil {
+				// Skipping because VPC endpoint routes are handled separately
+				// See aws_vpc_endpoint
+				continue
+			}
+
 			// Minimal data for route
 			d := subResource.Data(nil)
 			d.SetType("aws_route")

--- a/builtin/providers/aws/import_aws_route_table_test.go
+++ b/builtin/providers/aws/import_aws_route_table_test.go
@@ -38,7 +38,7 @@ func TestAccAWSRouteTable_importBasic(t *testing.T) {
 
 func TestAccAWSRouteTable_complex(t *testing.T) {
 	checkFn := func(s []*terraform.InstanceState) error {
-		// Expect 2: group, 1 rules
+		// Expect 3: group, 2 rules
 		if len(s) != 3 {
 			return fmt.Errorf("bad states: %#v", s)
 		}

--- a/builtin/providers/aws/import_aws_route_table_test.go
+++ b/builtin/providers/aws/import_aws_route_table_test.go
@@ -35,3 +35,136 @@ func TestAccAWSRouteTable_importBasic(t *testing.T) {
 		},
 	})
 }
+
+func TestAccAWSRouteTable_complex(t *testing.T) {
+	checkFn := func(s []*terraform.InstanceState) error {
+		// Expect 2: group, 1 rules
+		if len(s) != 3 {
+			return fmt.Errorf("bad states: %#v", s)
+		}
+
+		return nil
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckRouteTableDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccRouteTableConfig_complexImport,
+			},
+
+			resource.TestStep{
+				ResourceName:     "aws_route_table.mod",
+				ImportState:      true,
+				ImportStateCheck: checkFn,
+			},
+		},
+	})
+}
+
+const testAccRouteTableConfig_complexImport = `
+resource "aws_vpc" "default" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_hostnames = true
+
+  tags {
+    Name = "tf-rt-import-test"
+  }
+}
+
+resource "aws_subnet" "tf_test_subnet" {
+  vpc_id                  = "${aws_vpc.default.id}"
+  cidr_block              = "10.0.0.0/24"
+  map_public_ip_on_launch = true
+
+  tags {
+    Name = "tf-rt-import-test"
+  }
+}
+
+resource "aws_eip" "nat" {
+  vpc                       = true
+  associate_with_private_ip = "10.0.0.10"
+}
+
+resource "aws_internet_gateway" "gw" {
+  vpc_id = "${aws_vpc.default.id}"
+
+  tags {
+    Name = "tf-rt-import-test"
+  }
+}
+
+variable "private_subnet_cidrs" {
+  default = "10.0.0.0/24"
+}
+
+resource "aws_nat_gateway" "nat" {
+  count         = "${length(split(",", var.private_subnet_cidrs))}"
+  allocation_id = "${element(aws_eip.nat.*.id, count.index)}"
+  subnet_id     = "${aws_subnet.tf_test_subnet.id}"
+}
+
+resource "aws_route_table" "mod" {
+  count  = "${length(split(",", var.private_subnet_cidrs))}"
+  vpc_id = "${aws_vpc.default.id}"
+
+  tags {
+    Name = "tf-rt-import-test"
+  }
+
+  depends_on = ["aws_internet_gateway.ogw", "aws_internet_gateway.gw"]
+}
+
+resource "aws_route" "mod-1" {
+  route_table_id         = "${aws_route_table.mod.id}"
+  destination_cidr_block = "0.0.0.0/0"
+  nat_gateway_id         = "${element(aws_nat_gateway.nat.*.id, count.index)}"
+}
+
+resource "aws_route" "mod" {
+  route_table_id            = "${aws_route_table.mod.id}"
+  destination_cidr_block    = "10.181.0.0/16"
+  vpc_peering_connection_id = "${aws_vpc_peering_connection.foo.id}"
+}
+
+resource "aws_vpc_endpoint" "s3" {
+  vpc_id          = "${aws_vpc.default.id}"
+  service_name    = "com.amazonaws.us-west-2.s3"
+  route_table_ids = ["${aws_route_table.mod.*.id}"]
+}
+
+### vpc bar
+
+resource "aws_vpc" "bar" {
+  cidr_block = "10.1.0.0/16"
+
+  tags {
+    Name = "tf-rt-import-test"
+  }
+}
+
+resource "aws_internet_gateway" "ogw" {
+  vpc_id = "${aws_vpc.bar.id}"
+
+  tags {
+    Name = "tf-rt-import-test"
+  }
+}
+
+### vpc peer connection
+
+resource "aws_vpc_peering_connection" "foo" {
+  vpc_id        = "${aws_vpc.default.id}"
+  peer_vpc_id   = "${aws_vpc.bar.id}"
+  peer_owner_id = "187416307283"
+
+  tags {
+    Name = "tf-rt-import-test"
+  }
+
+  auto_accept = true
+}
+`

